src/provisioning-ofono.c

```diff
@@ -1025,31 +1071,38 @@ static void check_interfaces(struct modem_data *modem, DBusMessageIter *iter)
 	if (has_gprs) {
 		if (get_gprs_properties(modem) < 0)
 			clean_provisioning();
-	} else
-		clean_provisioning();
+	} else {
+		LOG("Provisioning failed for modem path: %s",modem->path);
+	}
 
 }
-static void remove_modem()
+static void remove_modems()
```

Comment:
If it's a remove callback for the hashtable (looks like it is), then it's supposed to free the data (modem) being removed from the table (and nothing else).